tensorflow/python/ops/math_ops.py

```diff
@@ -541,6 +541,9 @@ def pow(x, y, name=None):
     A `Tensor`.
   """
   with ops.name_scope(name, "Pow", [x]) as name:
+    if 'int' in str(x.dtype):
+      tf.assert_negative(y,
```

Comment:
don't you mean tf.assert_positive here?
also, I don't think this strategy will work. You need to make sure that there is not any negative exponents for example.
for example this will hang but doesn't have a consistent tf.assert_positive() or negative()

I'd start by writing a set of unit tests you want to prevent from haing i.e.
```
sess.run(tf.pow([5,2],[-2,3]))
sess.run(tf.pow([5,2],[2,-3]))
sess.run(tf.pow([5,2],-3))
```